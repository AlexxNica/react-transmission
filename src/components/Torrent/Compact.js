import React, { Component } from 'react';
import CSSModules from 'react-css-modules';
import { inject, observer } from 'mobx-react';

import ProgressBar from './ProgressBar';

import styles from './styles';

@inject('view_store')
@observer
@CSSModules(styles)
class Compact extends Component {
  render() {
    const torrent = this.props.torrent;

    return (
      <div styleName='torrentCompact'>
        <div styleName='nameCompact'>
          {torrent.name}
        </div>
        <div styleName='detailsCompact'>
          Ratio: 0.41, â†‘ 50 kB/s
        </div>
        <div styleName='progressBarRowCompact'>
          <ProgressBar torrent={torrent} />
        </div>
      </div>
    );
  }
}

export default Compact;
