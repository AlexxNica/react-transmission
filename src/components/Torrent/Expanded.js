import React from 'react';
import { findDOMNode } from 'react-dom'
import CSSModules from 'react-css-modules';
import { inject, observer } from 'mobx-react';

import PureComponent from 'components/PureComponent';
import ContextMenu from 'components/ContextMenu';
import ProgressBar from './ProgressBar';

import { getPeerDetails, getProgressDetails } from './services';

import styles from './styles/index.css';

@inject('view_store')
@observer
@CSSModules(styles)
class Expanded extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      showContextMenu: false,
      position: {
        left: 0,
        top: 0,
      },
    };

    this.onContextMenu = this.onContextMenu.bind(this);
    this.toggleContextMenu = this.toggleContextMenu.bind(this);
    this.renderContextMenu = this.renderContextMenu.bind(this);
  }

  onContextMenu(event) {
    const { clientX, clientY } = event;

    event.preventDefault();
    this.toggleContextMenu({left: clientX, top: clientY});
  }

  toggleContextMenu(position) {
    this.setState({showContextMenu: true, position});
  }

  renderContextMenu() {
    const { position, showContextMenu } = this.state;

    return (
      <div ref='target' style={{position: 'absolute', visibility: 'hidden', ...position, left: position.left + 50}}>
        <ContextMenu
          show={showContextMenu}
          container={this}
          target={() => {
            return findDOMNode(this.refs.target);
          }}
          onHide={() => this.setState({showContextMenu: false})}
        >
          <ul styleName='torrentMenu'>
            <li styleName='torrentMenuItem'>Pause</li>
            <li styleName='torrentMenuItem'>Resume</li>
            <li styleName='torrentMenuItem'>Resume Now</li>
            <li styleName='torrentMenuSeparator' />
            <li styleName='torrentMenuItem'>Move to Top</li>
            <li styleName='torrentMenuItem'>Move Up</li>
            <li styleName='torrentMenuItem'>Move Down</li>
            <li styleName='torrentMenuItem'>Move to Bottom</li>
            <li styleName='torrentMenuSeparator' />
            <li styleName='torrentMenuItem'>Remove From List...</li>
            <li styleName='torrentMenuItem'>Trash Data & Remove From List...</li>
            <li styleName='torrentMenuSeparator' />
            <li styleName='torrentMenuItem'>Verify Local Data</li>
            <li styleName='torrentMenuItem'>Set Location...</li>
            <li styleName='torrentMenuSeparator' />
            <li styleName='torrentMenuItem'>Ask tracker for more peers</li>
            <li styleName='torrentMenuSeparator' />
            <li styleName='torrentMenuItem'>Select All</li>
            <li styleName='torrentMenuItem'>Deselect All</li>
          </ul>
        </ContextMenu>
      </div>
    );
  }

  render() {
    const torrent = this.props.torrent;

    return (
      <div
        styleName='torrent'
        onContextMenu={this.onContextMenu}
      >
          <div styleName='name'>
            {torrent.name}
          </div>
          <div styleName='peerDetails'>
            {getPeerDetails(torrent)}
          </div>
          <div styleName='progressBarRow'>
            <ProgressBar torrent={torrent} />
            <button></button>
          </div>
          <div styleName='progressDetails'>
            {getProgressDetails(torrent)}
          </div>
          {this.renderContextMenu()}
      </div>
    );
  }
}

export default Expanded;
