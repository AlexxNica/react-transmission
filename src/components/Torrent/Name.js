import React from 'react';

function Name({ torrent }) {
  return (
    <p>
      <strong>{torrent.name}</strong>
    </p>
  );
}

export default Name;
