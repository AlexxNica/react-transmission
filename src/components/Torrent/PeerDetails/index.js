import React from 'react';

function PeerDetails({ torrent }) {
  return (
    <p>
      <strong>{torrent.totalSize}</strong>
    </p>
  );
}

export default PeerDetails;
