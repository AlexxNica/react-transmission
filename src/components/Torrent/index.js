import React, { Component} from 'react';

class Torrent extends Component {
  render() {
    const torrent = this.props.torrent;

    return (
      <div style={{margin: '10px', border: '1px solid grey', padding: '5px'}}>
        <p>
          <strong>{torrent.name}</strong>
        </p>
        <p>
          <strong>{torrent.totalSize}</strong>
        </p>
        <progress max="100" value="80"></progress>
        <p>{torrent.status}</p>
        <p>Downloading ... ↓ {torrent.rateDownload} ↑ {torrent.rateUpload}</p>
      </div>
    );
  }
}

export default Torrent;
