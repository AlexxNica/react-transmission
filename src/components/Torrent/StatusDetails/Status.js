import React from 'react';
import { FormattedMessage } from 'react-intl';

import Torrent from 'stores/torrent';

const {
  STATUS_STOPPED,
  STATUS_CHECK_WAIT,
  STATUS_CHECK,
  STATUS_DOWNLOAD_WAIT,
  STATUS_DOWNLOAD,
  STATUS_SEED_WAIT,
  STATUS_SEED,
} = Torrent;

export default function({ torrent }) {
  switch (torrent.status) {
    case STATUS_CHECK_WAIT:
      return <FormattedMessage id='torrent.status.checkwait' defaultMessage='Queued for verification' />;
    case STATUS_CHECK:
      return <FormattedMessage id='torrent.status.check' defaultMessage='Verifying local data' />;
    case STATUS_DOWNLOAD_WAIT:
      return <FormattedMessage id='torrent.status.downloadwait' defaultMessage='Queued for download' />;
    case STATUS_DOWNLOAD:
      return <FormattedMessage id='torrent.status.download' defaultMessage='Downloading' />;
    case STATUS_SEED_WAIT:
      return <FormattedMessage id='torrent.status.seedwait' defaultMessage='Queued for seeding' />;
    case STATUS_SEED:
      return <FormattedMessage id='torrent.status.seed' defaultMessage='Seeding' />;
    case STATUS_STOPPED:
      return (
      torrent.isFinished
        ? <FormattedMessage id='torrent.status.finished' defaultMessage='Finished' />
        : <FormattedMessage id='torrent.status.paused' defaultMessage='Paused' />
      );
    case null:
    case undefined:
      return <FormattedMessage id='torrent.status.unknown' defaultMessage='Unknown' />;
    default:
      return <FormattedMessage id='torrent.status.error' defaultMessage='Error' />;
  }
}
