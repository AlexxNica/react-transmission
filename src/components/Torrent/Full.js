import React, { Component } from 'react';
import CSSModules from 'react-css-modules';
import { inject, observer } from 'mobx-react';
import autobind from 'autobind-decorator';

import ProgressBar from './ProgressBar';
import StatusButton from './StatusButton';
import StatusDetails from './StatusDetails';
import ProgressDetails from './ProgressDetails';

import styles from './styles/index.css';

function getPeerDetailsStyles(torrent) {
  if (torrent.hasErrors) {
    return `peerDetailsError`;
  }

  return 'peerDetails';
}

@inject('torrents_store')
@observer
@CSSModules(styles)
class Full extends Component {
  @autobind onToggleTorrent(torrentId) {
    if (this.props.torrent.isStopped) {
      this.props.torrents_store.start(torrentId);
      return;
    }

    this.props.torrents_store.stop(torrentId);
  }

  render() {
    const { torrent } = this.props;

    return (
      <div styleName='torrent'>
        <div styleName='name'>
          {torrent.name}
        </div>
        <div styleName={getPeerDetailsStyles(torrent)}>
          <StatusDetails torrent={torrent} />
        </div>
        <div styleName='progressBarRow'>
          <ProgressBar torrent={torrent} />
          <StatusButton torrent={torrent} onToggle={this.onToggleTorrent} />
        </div>
        <div styleName='progressDetails'>
          <ProgressDetails torrent={torrent} />
        </div>
      </div>
    );
  }
}

export default Full;
