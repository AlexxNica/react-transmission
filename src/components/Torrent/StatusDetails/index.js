import React from 'react';

import Seeding from './Seeding';
import Checking from './Checking';
import Downloading from './Downloading';
import Error from './Error';
import Status from './Status';

function StatusDetails({ torrent }) {
  if (torrent.hasErrors) {
    return <Error torrent={torrent} />;
  }

  if (torrent.isDownloading) {
    return <Downloading torrent={torrent} />;
  }

  if (torrent.isSeeding) {
    return <Seeding torrent={torrent} />;
  }

  if (torrent.isChecking) {
    return <Checking torrent={torrent} />;
  }

  return <Status torrent={torrent} />;
}

export default StatusDetails;
