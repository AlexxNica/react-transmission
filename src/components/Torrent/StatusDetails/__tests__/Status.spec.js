import React from 'react';
import createComponentWithIntl from 'test/util/createComponentWithIntl';

import Status from '../Status';

import Torrent from 'stores/torrent';

const {
  STATUS_STOPPED,
  STATUS_CHECK_WAIT,
  STATUS_CHECK,
  STATUS_DOWNLOAD_WAIT,
  STATUS_DOWNLOAD,
  STATUS_SEED_WAIT,
  STATUS_SEED,
} = Torrent;

test('Status check wait', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: STATUS_CHECK_WAIT}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Status check', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: STATUS_CHECK}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Status download wait', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: STATUS_DOWNLOAD_WAIT}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Status download', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: STATUS_DOWNLOAD}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Status seed wait', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: STATUS_SEED_WAIT}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Status seed', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: STATUS_SEED}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Status stopped finished', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: STATUS_STOPPED, isFinished: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Status stopped not finished', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: STATUS_STOPPED, isFinished: false}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Status unknown', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: null}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Status other', () => {
  const component = createComponentWithIntl(
    <Status torrent={{status: -999}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
