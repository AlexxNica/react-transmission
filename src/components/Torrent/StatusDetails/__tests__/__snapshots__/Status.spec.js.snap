exports[`test Status check 1`] = `
<span>
  Verifying local data
</span>
`;

exports[`test Status check wait 1`] = `
<span>
  Queued for verification
</span>
`;

exports[`test Status download 1`] = `
<span>
  Downloading
</span>
`;

exports[`test Status download wait 1`] = `
<span>
  Queued for download
</span>
`;

exports[`test Status other 1`] = `
<span>
  Error
</span>
`;

exports[`test Status seed 1`] = `
<span>
  Seeding
</span>
`;

exports[`test Status seed wait 1`] = `
<span>
  Queued for seeding
</span>
`;

exports[`test Status stopped finished 1`] = `
<span>
  Finished
</span>
`;

exports[`test Status stopped not finished 1`] = `
<span>
  Paused
</span>
`;

exports[`test Status unknown 1`] = `
<span>
  Unknown
</span>
`;
